pkg/controllers/nodeclaim/inplaceupdate/patch.go

```diff
@@ -82,6 +102,33 @@ func CalculateVMPatch(
 	return update
 }
 
+// Given AKS machine support PUT, but not PATCH, the AKS machine object will be patched directly, while the returning Machine object is just a tracker for logging purposes.
+func CalculateAKSMachinePatch(
+	options *options.Options,
+	nodeClaim *karpv1.NodeClaim,
+	nodeClass *v1beta1.AKSNodeClass,
+	patchingAKSMachine *armcontainerservice.Machine,
+) *armcontainerservice.Machine {
+	update := &armcontainerservice.Machine{}
```

Comment:
Why not just do away with `update` entirely and return `patchingAKSMachine` too?

Then in the log function you can either:
1. Extract the fields you know are patchable and log those (there aren't many).
2. Use a diffing library like `"github.com/google/go-cmp/cmp"` (usually used for tests but could be used for something like this too).
3. Just log the whole machine (bigger log but contains the information we want and avoids the need to track what specifically we updated).